Remove crash! macro

saulvaldelvira opened the pull request:
Remove the crash! and crash_if_error! macros, as discussed in #5487 .

I've replaced all crash_if_err! macro calls, by changing the return type of the functions it was called on. Instead of returning T, they return UResult\<T\>.
This way, instead of crashing, we can propagate the error.

The prompt_yes! macro also used crash_if_err!. In this case, I've replaced that with a call to show_if_err!.

All test pass, except test_df::test_type_option_with_file, which already failed before the changes.

NOTE: This is my first time contributing. Hope I've done everything alright.

cakebaker commented:
Can you please run `cargo fmt`? The CI also complains about some doc links pointing to code you removed: https://github.com/uutils/coreutils/actions/runs/12641963814/job/35236266280?pr=7084

saulvaldelvira commented:
> Can you please run `cargo fmt`? The CI also complains about some doc links pointing to code you removed: https://github.com/uutils/coreutils/actions/runs/12641963814/job/35236266280?pr=7084

Sorry, yesterday I forgot to run cargo fmt. I've fixed the formatting and removed the doc comments linking to the deleted macros.

saulvaldelvira commented on src/uucore/src/lib/macros.rs:
```diff
@@ -31,7 +31,6 @@
 //!   - From custom messages: [`crate::show_error!`]
 //! - Print warnings: [`crate::show_warning!`]
 //! - Terminate util execution
```
Done.

cakebaker commented:
Thanks for your PR :)
